### PR TITLE
[MPDX-8798] Populate settings sidebar with special family categories

### PR DESCRIPTION
**Pull request by canac**
## Description

Instead of using a static list, populate the settings sidebar with the categories in the "special" family. #1421 has already updated the settings page to render the tables for the special family's categories.

MPDX-8798

## Checklist:

- [x] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [x] I have applied the appropriate labels. (_Add the label "On Staging" to get the branch automatically merged into staging._)
- [x] I have requested a review from another person on the project

### CODE REVIEW

File: src/components/Reports/GoalCalculator/CalculatorSettings/SettingsSectionList.tsx [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+import { getFamilySections } from '../Shared/familySections';
 import { SectionList } from '../SharedComponents/SectionList';
 
 export const SettingsSectionList: React.FC = () => {
   const { t } = useTranslation();
+  const {
+    goalCalculationResult: { data },
+  } = useGoalCalculator();
+  const specialFamily =
+    data?.goalCalculation.specialFamily.primaryBudgetCategories;
```
Comment (dr-bizz) [inline src/components/Reports/GoalCalculator/CalculatorSettings/SettingsSectionList.tsx:13]: [Must-fix] Null access risk accessing nested property. Evidence: data?.goalCalculation.specialFamily.primaryBudgetCategories then uses data?.goalCalculation.specialFamily. Impact: Runtime error if specialFamily exists but primaryBudgetCategories doesn't. Fix: Check specialFamily directly. [Category: File]

File: src/components/Reports/GoalCalculator/CalculatorSettings/SettingsSectionList.tsx [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
 
   const sections = [
     { title: t('Information'), complete: false },
-    { title: t('Special Income'), complete: false },
-    { title: t('One-time Goals'), complete: false },
+    ...(specialFamily
```
Comment (dr-bizz) [inline src/components/Reports/GoalCalculator/CalculatorSettings/SettingsSectionList.tsx:17]: [Reuse] SettingsStep.tsx has hardcoded sections while this component uses dynamic data. Evidence: SettingsStep:23-28 hardcodes 'Special Income' and 'One-time Goals'. Impact: Inconsistency. Suggestion: Use getFamilySections in SettingsStep too. [Category: Reuse]

File: src/components/Reports/GoalCalculator/CalculatorSettings/SettingsSectionList.test.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { render } from '@testing-library/react';
+import { GoalCalculatorTestWrapper } from '../GoalCalculatorTestWrapper';
+import { SettingsSectionList } from './SettingsSectionList';
+
+describe('SettingsSectionList', () => {
+  it('renders correctly', async () => {
```
Comment (dr-bizz) [inline src/components/Reports/GoalCalculator/CalculatorSettings/SettingsSectionList.test.tsx:6]: [Suggested tests] Could we add test for null/undefined specialFamily data to verify null safety? Evidence: Component accesses nested properties. Impact: Edge case coverage. [Category: File]

File: src/components/Reports/GoalCalculator/CalculatorSettings/SettingsSectionList.tsx (hunk 2 above)
Comment (canac) [reply src/components/Reports/GoalCalculator/CalculatorSettings/SettingsSectionList.tsx:17]: You're right that `SettingsStep` needs to be changed, but Will is already doing that in his PR: https://github.com/CruGlobal/mpdx-react/pull/1421/files#diff-36afeff2b25aa3047f1c771d61efb8fe34c92da5e19fb8f524ccc90167305852.
